### DISCUSSION TITLE
Hard bounces

### DISCUSSION THREAD

**Discussion by rbustos** [postalserver/postal#2547] [Help with using Postal]
Hard Bounces detected by postal (at 1st try) will be sent to suppression list immediately? 

I did some tests by sending non existing user account in one of my domains and even when they all fails and are marked as hard bounce. They are continuously being sent to the remote MTA. This is not what I should expect once a hard bounce is logged since it should stop sending new ones to the same destination. 

I'm using following config:
  maximum_delivery_attempts: 7
  maximum_hold_expiry_days: 7
  suppression_list_removal_delay: 30

Any idea? maybe some config that I've missed.

Thanks in advance

**Comment by willpower232** [comment]
Unfortunately the nature of email means that almost all issues are temporary if you trust your users to enter the correct email address to send to so Postal leans heavily on retrying a message as often as makes sense/configured.

You can see here the bit where it decides to give up on sending a message.

https://github.com/postalserver/postal/blob/2f62baa238fc1102706ee4acf079b7a876b05283/app/jobs/unqueue_message_job.rb#L55-L73

**Comment by jshapiro2** [comment]
Why would you not have something like the following auto hard bounce and not try to resend
  "details": "No SMTP servers were available for bounce-testing.postmarkapp.com. Tried bounce-testing.postmarkapp.com",

**Reply by willpower232** [reply]
> Unfortunately the nature of email means that almost all issues are temporary

there is no way of Postal telling the difference between "my server has lost internet", "someone changed the DNS and broke it", and "this DNS has never been correct" so it is better to err on the side of caution and keep retrying with bigger delays until the maximum setting is reached.
